src/pages/create-or-edit-lesson/CreateOrEditLesson.tsx

```diff
@@ -165,44 +165,53 @@ const CreateOrEditLesson = () => {
   } = useForm<LessonData>({
     initialValues,
     validations,
-    onSubmit: id ? fetchEditedLesson : fetchAddLesson,
+    onSubmit: () => {
+      if (id) {
+        fetchEditedLesson()
+      } else {
+        fetchAddLesson()
+      }
+    },
     submitWithData: true
   })
 
-  const getLesson = (id?: string): Promise<AxiosResponse> => {
+  const getLesson = (id?: string) => {
     return ResourceService.getLesson(id)
   }
```

Comment:
You can use closures here to encapsulate passing id to this function. It will help you to pass just reference to this function in the `queryFn` prop
```suggestion
  const getLesson = useCallback(() => {
    return ResourceService.getLesson(id)
  }, [id])
```